rest/context.go

```diff
@@ -325,6 +351,60 @@ func (ctx *gorillaRequestContext) NextURL() (string, error) {
 	return u.String(), nil
 }
 
+func (ctx *gorillaRequestContext) buildURL(fullPath bool, resourceName string,
+	methodName string, pairs ...string) (string, error) {
+	r, ok := ctx.Request()
+	if !ok {
+		return "", fmt.Errorf("unable to build URL for resource name %q: no request available.",
+			resourceName)
+	}
+
+	routeName := resourceName + ":" + methodName
+	route := ctx.router.Get(routeName).Host(r.Host)
+
+	var builder func(pairs ...string) (*url.URL, error)
+	if fullPath {
+		builder = route.URL
+	} else {
+		builder = route.URLPath
+	}
+
+	pairs = append(pairs, "version", ctx.Version())
+	url, err := builder(pairs...)
+	if err != nil {
+		return "", err
+	}
+
+	if r.TLS != nil {
+		url.Scheme = "https"
+	}
+
+	return url.String(), nil
+}
+
+// BuildURL builds a full URL for the given resource name & method name with the
+// key/value pairs for the route variables.
+//
+// resourceName should have the same value as the handler's ResourceName method.
+//
+// methodName should be the Handle* constant that corresponds with the resource
+// method for which to build the URL. E.g. HandleCreate with build a URL that
+// corresponds with the CreateResource method.
+//
+// pairs should be key/value pairs for any URL variables, passed on to
+// gorilla's Route.URL method.
```

Comment:
I hate, *hate*, **hate** any duplication in source code. I'd rather the comment on the implementation be deleted rather than copy/pasted from the interface.

@tylertreat-wf thoughts?